testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/ejb/RemoteEJBDirectory.java

```diff
@@ -29,6 +29,12 @@
 
 /**
  * {@link EJBDirectory} that uses remote JNDI.
+ *
+ * NOTE:
+ * if you hold a static reference to this class, it causes any JNDI lookups across different tests to use the same discovered node registry (DNR).
+ * This can cause server starts and stops in one test to contaminate other tests and produce incorrect results.
+ * It is adviseable to use one instance per test by defining a @Before and @After method to create and dispose of the instance on a per test basis.
```

Comment:
Generally, it is inadvisable to hold _any_ references to an EJBDirectory.
The preferred pattern of usage is:
```java
for (EJBDirectory directory = new RemoteEJBDirectory(MODULE_NAME)) {
   // lookup EJBs
   // invoke EJBs
   // remove EJBs
}
```